### PR TITLE
Add mocks for unit tests

### PR DESCRIPTION
**Pull request by gsingh93**

### CODE REVIEW

File: unit-tests/mocks/gdblib.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+from unittest.mock import MagicMock
+
+from mocks.arch import Amd64Arch
+from mocks.config import Config
+from mocks.typeinfo import Amd64TypeInfo
+
+
+class GdbLib(types.ModuleType):
+    def __init__(self, module_name):
+        super(GdbLib, self).__init__(module_name)
```
Comment (disconnect3d) [inline unit-tests/mocks/gdblib.py]:
```suggestion
        super().__init__(module_name)
```

**Review by disconnect3d — APPROVED** [review]
lgtm; one note: maybe we should move unit-tests to tests/?

**Comment by gsingh93** [comment]
> lgtm; one note: maybe we should move unit-tests to tests/?

I was thinking of we should move the content of `tests` to `tests/gdb-tests` and have the unit tests be in `tests/unit-tests`. We can't have `unit-tests` in `tests` without moving the existing tests to some other folder, as `conftest.py` gets imported, which we don't want for `unit-tests`. If this directory structure sounds good, I'll update this PR with it.

**Comment by disconnect3d** [comment]
Sounds good to me. Btw we can have multiple conftest.py files afaik

File: tests/unit-tests/mocks/gdblib.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import sys
+import types
+from unittest.mock import MagicMock
```
Comment (CptGibbon) [inline tests/unit-tests/mocks/gdblib.py:3]: Do you think mixing `unittest` and `pytest` will be okay?

Comment (gsingh93) [reply tests/unit-tests/mocks/gdblib.py:3]: Yup, pytest is designed to work well with `unittest`

**Pull request merged**